test/utils/cluster.go

```diff
@@ -0,0 +1,113 @@
+// Copyright 2023 Google LLC
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//      http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package utils
+
+import (
+	"bytes"
+	"fmt"
+	"os/exec"
+	"strconv"
+
+	"k8s.io/klog/v2"
+)
+
+func EnsureCluster(location, clusterName string, numOfNodes int) error {
+	params := []string{
+		"container",
+		"clusters",
+		"describe",
+		clusterName,
+		"--zone", location,
+	}
+	out, err := exec.Command("gcloud", params...).CombinedOutput()
+	if err != nil {
+		klog.Infof("Cluster %s does not exist, creating instead.", clusterName)
+		return createCluster(location, clusterName, numOfNodes)
+	}
+
+	pattern := "currentNodeCount: "
```

Comment:
Break these string parsing routines into their own methods and have unit tests for these.

Also, this seems to be quite complex, doesn't a regex search work for this, or modifying the output of the `gcloud` command to print only the fields you need:

https://cloud.google.com/sdk/gcloud/reference/topic/formats 

search for "EXAMPLES"

Reply:
Updated.